source/user-management.rst

```diff
@@ -0,0 +1,220 @@
+
+.. note:: The passwords are stored in plain text as `stringData <https://kubernetes.io/docs/concepts/configuration/secret/#creating-a-secret-manually>`_ which is converted by Kubernetes to base64 on commit.  Depending on how you retrieve the data later this may need to be unencoded.
+
+The Operator automatically tracks changes in the ``stringData`` field, if any.
+
+Valid Privileges for Automation
+*******************************
+
+This methodology allows the privileges field to be free-form. All valid privileges listed `in the official MySQL documentation <https://dev.mysql.com/doc/refman/8.0/en/grant.html>`_ are supported.
+
+The Operator **does not support** the ``AS user``, ``WITH GRANT OPTION``, or ``WITH ADMIN OPTION`` functionality, so this limits some of the use cases for specific privileges such as ``PROXY``.
+
+Managing and Mapping MySQL Roles
+********************************
+
+Roles are defined in the *roles dictionary* within the YAML object. Each role has a set of grants associated with it and a defined name. 
+
+.. note:: Instead of adding a privileges key to the *users dictionary*, you can specify the roles key, and that role will be added to that user by generating a series of one or more ``GRANT role TO user`` statements.
+
+Using Kubernetes Selectors in Your Application
+**********************************************
+
+You can make use of Selectors to reference the content of the secret object within your application's Pods running in the same Kubernetes namespace assuming appropriate `RBAC rules <https://kubernetes.io/docs/reference/access-authn-authz/rbac/>`_ are set.
+
+Within your application Pod configuration, you may have something like the following.
+
+.. code:: yaml
+
+   spec:
+     containers:
+       env:
+     - name: SECRET_PASSWORD
+       valueFrom:
+         secretKeyRef:
+           name: secrets-for-users
+           key: myuser1
+
+By doing this, your application Pod has an environment variable set named ``SECRET_PASSWORD`` which contains the value of the password for the specified user.  You can then use this in your application to build your connection string to the database.
+
+Password Rotation Policies and Timing
+*************************************
+
+Password rotation is performed within moments of the password being changed in the Secrets object. The reason for this is that ProxySQL does not currently support dual passwords, which is a feature added in MySQL 8.0.14.  As such, it's imperative that you are prepared to update your application connection strings shortly after making the necessary password change for a user in the Secrets object.
+
+How Method 1 and Method 2 Interact
+----------------------------------
+
+When the Operator detects that user management must happen it generates a single transaction that contains the following:
+
+* one or more ``DROP USER IF EXISTS`` statements,
+* one or more ``CREATE USER`` statements,
+* one or more `GRANT` statements.
```

Comment:
GRANT should have two sets of backticks in order to get appropriately formatted in the output.